esp-hal/src/dma/gdma.rs

```diff
@@ -170,6 +170,29 @@ impl RegisterAccess for AnyGdmaTxChannel {
         self.ch()
             .out_link()
             .modify(|_, w| w.outlink_start().set_bit());
+
+        // Delay until there is data in the DMA's out buffer
+        cfg_if::cfg_if! {
+            if #[cfg(esp32s3)] {
+                while self
+                    .ch()
+                    .outfifo_status()
+                    .read()
+                    .outfifo_empty_l3()
+                    .bit_is_set()
+                {}
+            } else if #[cfg(any(esp32c6, esp32h2))] {
+                // FIXME: For some reason, C6 and H2 may loop forever here
+            } else {
+                while self
+                    .ch()
+                    .outfifo_status()
+                    .read()
+                    .outfifo_empty()
+                    .bit_is_set()
+                {}
+            }
+        }
```

Comment:
Rather than hijack the `start` method here, could this be a `TxRegisterAccess::is_fifo_empty()` then the loop can live in `ChannelTx::start_transfer`.

Reply:
If we'd strip this out (which comes at a cost - we have to remember to call that function), we might as well expose it to the peripheral drivers and let them wait at the last possible moment. Right now, however, I prefer the safety of the current code.

Reply:
> we have to remember to call that function

You only have to "remember" to do it here once right? https://github.com/esp-rs/esp-hal/blob/04ba779f2d41192ddd412a0d1bc35957889cb3a5/esp-hal/src/dma/mod.rs#L2311

I'm not sure what you mean